draft-irtf-cfrg-bbs-signatures.md

```diff
@@ -410,20 +410,19 @@ Deserialization:
 
 Procedure:
 
-1.  (Q_1, Q_2, H_1, ..., H_L) = create_generators(L+2)
-2.  domain = calculate_domain(PK, Q_1, Q_2, (H_1, ..., H_L), header)
+1.  (Q_2, H_1, ..., H_L) = create_generators(L+1)
+2.  domain = calculate_domain(PK, Q_2, (H_1, ..., H_L), header)
 3.  if domain is INVALID, return INVALID
-4.  e_s_octs = serialize((SK, domain, msg_1, ..., msg_L))
-5.  if e_s_octs is INVALID, return INVALID
-6.  e_s_len = octet_scalar_length * 2
-7.  e_s_expand = expand_message(e_s_octs, expand_dst, e_s_len)
-8.  if e_s_expand is INVALID, return INVALID
-9.  e = hash_to_scalar(e_s_expand[0..(octet_scalar_length - 1)])
-10. s = hash_to_scalar(e_s_expand[octet_scalar_length..(e_s_len - 1)])
-11. if e or s is INVALID, return INVALID
-12. B = P1 + Q_1 * s + Q_2 * domain + H_1 * msg_1 + ... + H_L * msg_L
+4.  e_octs = serialize((SK, domain, msg_1, ..., msg_L))
+5.  if e_octs is INVALID, return INVALID
+6.  e_len = octet_scalar_length
+7.  e_expand = expand_message(e_octs, expand_dst, e_len)
+8.  if e_expand is INVALID, return INVALID
+9.  e = hash_to_scalar(e_expand)
+11. if e is INVALID, return INVALID
+12. B = P1 + Q_2 * domain + H_1 * msg_1 + ... + H_L * msg_L
 13. A = B * (1 / (SK + e))
```

Comment:
```suggestion
4. e = hash_to_scalar(serialize((SK, domain, msg_1, ..., msg_L)))
5. if e is INVALID, return INVALID
6. B = P1 + Q_2 * domain + H_1 * msg_1 + ... + H_L * msg_L
7. A = B * (1 / (SK + e))
```

Not sure if a step checking that `serialize((SK, domain, msg_1, ..., msg_L))` is VALID is needed. If the inputted arguments of Sign are correct, all values passed to `serialize` are scalars that should not cause serialize to abort.

Probably should add `L` in there, but this can be done in another PR. Even if we do though, `serialize` will fail only if `L > 2^64 - 1`, in which case `calculate_domain` would fail at step 2.